src/library_memfs.js

```diff
@@ -382,17 +380,14 @@ mergeInto(LibraryManager.library, {
             }
           }
           allocated = true;
-          // malloc() can lead to growing the heap. If targeting the heap, we need to
```

Comment:
Just out of curiosity. Do we know what this was about?

In my music projects (using AssemblyScript, not emscripten) I've experienced that the heap arraybuffer becomes "neutered" after growth, and so I can't just keep references in JS, they'll have to be re-created. Was it something similar behind `re-acquire the heap buffer object` as written here? And is this something we don't have to worry about anymore?

Reply:
Basically yes.  IIUC, HEAD8 and other globals that refer to the memory buffer get automatically.

See `updateGlobalBufferAndViews` in `src/preamble.js`.  I think the problem occurs if you assign HEAP8 to some other variable.. that variable can then point to the stale buffer,.